### ISSUE TITLE
Necessary improvements on the UI of tuplets

### ISSUE THREAD

**Issue by mike-spa** [musescore/MuseScore#18980]
### Your idea

<img width="364" alt="image" src="https://github.com/musescore/MuseScore/assets/93707756/3217819c-c4fb-4f8d-8cd9-546519e26ff7">

At the moment, it is not possible to create this tuplet in Musescore. This is a very silly limitation, considering that the screenshot above is obtained... _in Musescore_.

I have realized this example by writing the tuplet manually into our file, to prove that our system is already perfectly capable of representing and handling such tuplet. The problem is a purely UI one, meaning that we don't have a flexible enough UI to allow creating it.

### Problem to be solved

In Musescore, a tuplet can only be created by selecting a _single_ rythmic figure and turning it into a tuplet. The natural consequence is that we can create tuplets only for those durations that _can_ be represented by a single figure. Consider this example.

I can easily create a 3:4 tuplet inside the 9-tuplet, because I can compress the last 4 quavers into a single 2/4 figure. However, I can't create a 3:5 tuplet, because there is no single figure which can represent a duration of 5 quavers.

https://github.com/musescore/MuseScore/assets/93707756/b15a694e-3171-43d9-9ac4-bc6a2fafe96f

I think that the most naive solution here is just to change the UI so that the tuplet isn't entered on the duration _of the figure_, but on the duration _of the selected range_. This way, the tuplet can be entered into any duration, not just the ones representable by a single musical figure. In principle I think it could be done quite easily. However, maybe the whole tuplets UI needs some design consideration.

**Comment by mmeyn** [comment]
This might look like a very special case because of the nested tuplets but this problem is there also for simpler cases: In 5/4 time,
![Bildschirmfoto vom 2023-08-09 16-21-56](https://github.com/musescore/MuseScore/assets/7814361/46cb01ac-871c-418b-a985-f388d59c6c25)
a whole measure rest can be divided into a quartuplet because it is represented by a single symbol,
![Bildschirmfoto vom 2023-08-09 16-21-52](https://github.com/musescore/MuseScore/assets/7814361/3fa86d7b-1420-4245-bfab-e56627e6171f)
but selecting a range the length of 5/8 cannot be divided into a single tuplet but divides all (in this case both) parts in to a quartuplet:
![Bildschirmfoto vom 2023-08-09 16-21-49](https://github.com/musescore/MuseScore/assets/7814361/9ae5ae97-eb9f-4036-8188-6b3efa7fb1d0)

**Comment by mike-spa** [comment]
Yep, that is entirely correct. The reason I presented it with nested tuplets is that your example is _technically_ possible to achieve in Musescore, albeit with an extremely silly workaround:
- Create a measure with a duration corresponding to the full duration of the tuplet.
- Make sure it has a full measure rest.
- Enter the tuplet you need on the full measure rest.
- Copy-paste the tuplet where you needed it.

The problem is that copy-paste won't work inside another tuplet, so while writing your examples is _technically_ possible, mine simply isn't, even with the workaround.

But yes, there 100% should be a better UI design for all of these cases.

**Comment by mmeyn** [comment]
If this leads to a UI redesign of tuplet input, I’d like to add a feature request: I’d very much like to input this rhythm
![grafik](https://github.com/musescore/MuseScore/assets/7814361/e82489ef-cd63-4c83-a274-5608b2b763bf)
as “N 5 Ctrl+3 C D E F E D E F G A G F” instead of the currently necessary “N 5 Ctrl+3 C D E 5 Ctrl+3 F E D 5 Ctrl+3 E F G 5 Ctrl+3 A G F”.

There are two things to consider here:
1. Selecting quarter duration (by pressing 5) and dividing it into a triplet changes the duration to eighths (as if you had pressed 4). This is nice but for continuing the triplet rhythm one has to reset the input duration to quarter (5) after the tuplet is completed because otherwise the next Ctrl+3 will divide an eighth into three sixteenths.
2. Having to press Ctrl+3 every time is not ideal. But if you just stayed in eighth triplet instead of returning to regular eighths or quarters there would be a need for “leave tuplet mode”. Simply pressing 4 or 5 wouldn’t be enough because you might want those inside of tuplet mode for rhythms like
![grafik](https://github.com/musescore/MuseScore/assets/7814361/f636edac-da1d-422c-a75b-4a3c32c43942)

**Comment by mmeyn** [comment]
> Yep, that is entirely correct. The reason I presented it with nested tuplets is that your example is _technically_ possible to achieve in Musescore, albeit with an extremely silly workaround:

OK, I understand why you did that. An inexperienced user like me probably wouldn’t find such a workaround on their own …

**Comment by Tantacrul** [comment]
I also agree with @mmeyn that perhaps activating a tuplet while in Note Input mode keeps using tuplets until you specify that you want to stop. Not 100% sure of the possible knock-on consequences of this though. It would need to be prototyped.

**Comment by MarcSabatella** [comment]
It would need a UI *if* we wanted to make this capability available within note input mode, where tuplets are normally entered.  I could imagine something pretty simple though - the custom tuplet dialog would have a couple of spinboxes to set custom duration the same way we do for measure duration.  It would default to whatever is currently selected on the note input toolbar.

But indeed, if we limit it to just normal mode, it could be made to work on a range selection with no additional UI component.

As for "sticky tuplets", it's a common request indeed and has been given thought over the years.  Clearly, for music where there are a ton of tuplets - eg, Moonlight Sonata - it's very useful.  For cases where there are just occasional triplets here and there in a string of eighth notes - eg, most jazz lines - it ends up being extra work to have to explicitly leave tuplet mode.  But, given that in the latter case we just established it's just occasional, I don't think it too unreasonable to need an extra keystroke.

**Comment by XiaoMigros** [comment]
For what it's worth: I created a plugin to tackle this specific problem, see https://musescore.org/en/project/advanced-tuplets
I think that while the UI isn't that great it has all the available options one might need for creating more complex tuplets.

However, it doesn't seem to work for tuplets not currently supported, it causes corruptions when creating them. So I wonder if an issue would arise in the creation process of such tuplets, even if they are technically supported by hardcoding them into a mscx file.

Also, @MarcSabatella, do you remember the issue we encountered when trying to create a tuplet with a different note durations to the ratio that MuseScore calculates for us? I think it was a duplet over a dotted quarter rest, MuseScore always made 2 eighths but sometimes one might want 2 quarter notes instead.

**Comment by MarcSabatella** [comment]
I remember that we discussed something, but I don't recall what.  There are definitely limitations along the lines of what you mention, though.

**Comment by Jojo-Schmitz** [comment]
Came up again in https://musescore.org/de/node/375557 (a 8:5 and a 7:5 tuplet in a composition of Aribert Reimann)
